pkg/orchestrator/scheduler/batch_job.go

```diff
@@ -0,0 +1,202 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/bacalhau-project/bacalhau/pkg/jobstore"
+	"github.com/bacalhau-project/bacalhau/pkg/lib/math"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/orchestrator"
+	"github.com/bacalhau-project/bacalhau/pkg/requester"
+	"github.com/bacalhau-project/bacalhau/pkg/util/generic"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"github.com/samber/lo"
+)
+
+// BatchJobScheduler is a scheduler for batch jobs that run until completion
+type BatchJobScheduler struct {
+	jobStore       jobstore.Store
+	planner        orchestrator.Planner
+	nodeDiscoverer requester.NodeDiscoverer
+	nodeRanker     requester.NodeRanker
+}
+
+type BatchJobSchedulerParams struct {
+	JobStore       jobstore.Store
+	Planner        orchestrator.Planner
+	NodeDiscoverer requester.NodeDiscoverer
+	NodeRanker     requester.NodeRanker
+}
+
+func NewBatchJobScheduler(params *BatchJobSchedulerParams) *BatchJobScheduler {
```

Comment:
Are we planning to add anything to the Job model to identify what type of job it is (batch, system, cron etc)?

Reply:
Yes, eventually. There are two options to consider https://www.notion.so/expanso/Modular-Architecture-In-Depth-2bd19c8acbe046c298319045edde18af

Still though, my next plan is to introduce a new internal job type that is different than we expose to end users through the CLI or APIs.